gfe_integration_tests/sdram_edge_tests/common/src/sdram.c

```diff
@@ -0,0 +1,213 @@
+
+//! The current timer tick value.
+// the timer tick callback returning the same value.
+uint32_t time;
+
+//! timer tick period (in microseconds)
+static uint32_t timer_period;
+
+//! The number of timer ticks to run for before being expected to exit
+static uint32_t simulation_ticks = 0;
+
+//! Determines if this model should run for infinite time
+static uint32_t infinite_run;
+
+//! read counter
+static uint32_t reads = 0;
+
+//! \brief Initialises the model by reading in the regions and checking
+//!        recording data.
+//! \return True if it successfully initialised, false otherwise
+static bool initialise(void) {
+    log_debug("Initialise: started");
+
+    // Get the address this core's DTCM data starts at from SRAM
+    data_specification_metadata_t *ds_regions =
+            data_specification_get_data_address();
+
+    // Read the header
+    if (!data_specification_read_header(ds_regions)) {
+        return false;
+    }
+
+    // Get the timing details and set up the simulation interface
+    if (!simulation_initialise(
+            data_specification_get_region(SYSTEM, ds_regions),
+            APPLICATION_NAME_HASH, &timer_period, &simulation_ticks,
+            &infinite_run, &time, SDP, DMA)) {
+        return false;
+    }
+
+    // HANDLE SDRAM EDGE STUFF
+    out_data = (sdram_regions_t*) data_specification_get_region(
+        SDRAM_OUT, ds_regions);
+    in_data = (sdram_regions_t*) data_specification_get_region(
+        SDRAM_IN, ds_regions);
+
+    // init the sdram to zeros
+    for (uint32_t region_id = 0; region_id < out_data->n_regions;
+            region_id++) {
+        for (uint32_t words_index = 0;
+                words_index < (uint32_t) out_data->regions[region_id].size / 4;
+                words_index++) {
+            out_data->regions[region_id].base_address[words_index] = 0;
+        }
+    }
+
+    log_debug("Initialise: finished");
+    return true;
+}
+
+//! \brief the function to call when resuming a simulation
+void resume_callback(void) {
+}
+
+//! \brief Timer interrupt callback
+//! \param[in] timer_count: the number of times this call back has been
+//!            executed since start of simulation
+//! \param[in] unused: unused parameter kept for API consistency
+void timer_callback(UNUSED uint timer_count, UNUSED uint unused) {
+    time++;
+    log_debug("Timer tick %u \n", time);
+
+    /* if a fixed number of simulation ticks that were specified at startup
+     * then do reporting for finishing */
+    if (infinite_run != TRUE && time >= simulation_ticks) {
+
+        // Enter pause and resume state to avoid another tick
+        simulation_handle_pause_resume(resume_callback);
+
+        log_debug("Completed a run");
+
+        // Subtract 1 from the time so this tick gets done again on the next
+        // run
+        time--;
+
+        simulation_ready_to_read();
+        return;
+    }
+
+    // if odd. add 1 to all out regions. else read ins and check count matches
```

Comment:
I would probably change the order of this comment so that it says "if even, read ins and check count matches, else add 1 to all out regions", since that's the order you've done it in the code.

Reply:
fixed